modules/programs/aerospace.nix

```diff
@@ -48,27 +48,87 @@ in
       type = types.submodule {
         freeformType = tomlFormat.type;
         options = {
-          start-at-login = lib.mkOption {
-            type = types.bool;
+          start-at-login = mkOption {
+            type =
+              with types;
+              oneOf [
+                bool
+                (submodule {
+                  options = {
+                    enable = lib.mkEnableOption "the AeroSpace launchd service";
+                    keepAlive = mkOption {
+                      type = bool;
+                      default = true;
+                      description = ''
+                        If set to true, launchd will automatically restart AeroSpace
+                        if it crashes or is quit for any reason.
+                      '';
+                    };
+                  };
+                })
+              ];
             default = false;
-            description = "Start AeroSpace at login.";
+            example = lib.literalExpression "true";
+            description = ''
+              Configure the launchd agent to manage the AeroSpace process.
+
+              Example (simple): `start-at-login = true;`
+              Example (advanced): `start-at-login = { enable = true; keepAlive = false; };`
+
+              The first time this is enabled, macOS will prompt you to allow this background
+              item in System Settings.
+
+              You can verify the service is running correctly from your terminal.
+              Run: `launchctl list | grep aerospace`
+
+              - A running process will show a Process ID (PID) and a status of 0, for example:
+                `12345	0	org.nix-community.home.aerospace`
+
+              - If the service has crashed or failed to start, the PID will be a dash and the
+                status will be a non-zero number, for example:
+                `-	1	org.nix-community.home.aerospace`
+
+              In case of failure, check the logs with `cat /tmp/aerospace.err.log`.
+              For more detailed service status, run `launchctl print gui/$(id -u)/org.nix-community.home.aerospace`.
+            '';
           };
           after-login-command = mkOption {
             type = with types; listOf str;
             default = [ ];
             description = ''
-              You can use it to add commands that run after login to macOS user session.
-              'start-at-login' needs to be 'true' for 'after-login-command' to work.
+              DEPRECATED: Commands from this option are now automatically merged into `after-startup-command`.
+
+              Please use `programs.aerospace.userSettings.after-startup-command` directly for all startup commands.
+
+              This option is kept for backward compatibility. It no longer has any effect on the launchd service itself.
+              All commands provided here will be executed by AeroSpace via its configuration file.
```

Comment:
```suggestion
```

The option wasn't deprecated upstream, we shouldn't say it is a deprecated option.